bigbluebutton-html5/imports/ui/components/presentation/component.jsx

```diff
@@ -1,78 +1,248 @@
+    let clientHeight;
+    let clientWidth;
+    if (this.props.userIsPresenter) {
+      clientHeight = whiteboardSizeAvailable.clientHeight;
+      clientWidth = whiteboardSizeAvailable.clientWidth;
+    } else {
+      clientHeight = presentationPaper.clientHeight;
+      clientWidth = presentationPaper.clientWidth;
+    }
+
+    // setting the state of the paperWidth and paperheight (available space for the svg)
+    // and set the showSlide to true to start rendering the slide
+    this.setState({
+      paperHeight: clientHeight,
+      paperWidth: clientWidth,
+      showSlide: true,
+    });
+  }
+
+  handleResize() {
+    const { presentationPaper, whiteboardSizeAvailable } = this;
+
+    if (presentationPaper) {
+      // if a user is a presenter - this means there is a whiteboardToolBar on the right
+      // and we have to get the width/height of the whiteboardSizeAvailable
+      // (inner hidden div with absolute position)
+      let clientHeight;
+      let clientWidth;
+      if (this.props.userIsPresenter) {
+        clientHeight = whiteboardSizeAvailable.clientHeight;
+        clientWidth = whiteboardSizeAvailable.clientWidth;
+      // user is not a presenter - we can get the sizes of the presentationPaper
+      // direct parent of the svg wrapper
+      } else {
+        clientHeight = presentationPaper.clientHeight;
+        clientWidth = presentationPaper.clientWidth;
+      }
+
+      // updating the size of the space available for the slide
+      this.setState({
+        paperHeight: clientHeight,
+        paperWidth: clientWidth,
+      });
+    }
   }
 
+  calculateSize() {
+    const originalWidth = this.props.currentSlide.width;
+    const originalHeight = this.props.currentSlide.height;
+
+    let adjustedWidth;
+    let adjustedHeight;
+
+    // Slide has a portrait orientation
+    if (originalWidth <= originalHeight) {
+      adjustedWidth = (this.state.paperHeight * originalWidth) / originalHeight;
+      if (this.state.paperWidth < adjustedWidth) {
+        adjustedHeight = (this.state.paperHeight * this.state.paperWidth) / adjustedWidth;
+        adjustedWidth = this.state.paperWidth;
+      } else {
+        adjustedHeight = this.state.paperHeight;
+      }
+
+      // Slide has a landscape orientation
+    } else {
+      adjustedHeight = (this.state.paperWidth * originalHeight) / originalWidth;
+      if (this.state.paperHeight < adjustedHeight) {
+        adjustedWidth = (this.state.paperWidth * this.state.paperHeight) / adjustedHeight;
+        adjustedHeight = this.state.paperHeight;
+      } else {
+        adjustedWidth = this.state.paperWidth;
+      }
+    }
+    return {
+      width: adjustedWidth,
+      height: adjustedHeight,
+    };
+  }
+
+  // renders the whole presentation area
   renderPresentationArea() {
-    let slideObj = this.props.currentSlide;
+    // sometimes tomcat publishes the slide url, but the actual file is not accessible (why?)
```

Comment:
You find the reason or open a issue about this?

Reply:
It's a weird bug, I wasn't sure if it was my own issue. It happened to me exclusively after I didn't turn off my VM for a couple of weeks, but then @KDSBrowne and @mmarangoni saw it once or twice.

I haven't seen it after I switched to 2.0, either it's fixed itself, or I got more memory on a new machine. It's difficult to reproduce.

Added a check here just in case, if I see this happening again - I'll definitely open an issue.